src/main/java/teammates/ui/controller/AdminEmailTrashAction.java

```diff
@@ -17,7 +17,7 @@ protected ActionResult execute() {
         String emailId = getRequestParamValue(Const.ParamsNames.ADMIN_EMAIL_ID);
         
         String redirect = getRequestParamValue(Const.ParamsNames.ADMIN_EMAIL_TRASH_ACTION_REDIRECT);
-        redirect = redirect == null ? Const.ActionURIs.ADMIN_EMAIL_TRASH_PAGE : redirect;
+        redirect = redirect == null ? Const.ActionURIs.ADMIN_EMAIL_TRASH_PAGE : redirect; // NOPMD
```

Comment:
PMD is giving a false positive here. Related to https://sourceforge.net/p/pmd/bugs/1340/, I think. It will be fixed in the next PMD version.